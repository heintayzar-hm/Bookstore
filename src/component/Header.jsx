import React from 'react';
import { NavLink } from 'react-router-dom';

// eslint-disable-next-line react/prefer-stateless-function
class Header extends React.Component {
  constructor(props) {
    super();
    this.props = props;
  }

  render() {
    const links = [
      {
        id: 1,
        path: '/books',
        text: 'Books',
      },
      {
        id: 2,
        path: '/categories',
        text: 'Categories',
      },
    ];
    return (
      <nav>
        <h1>BookStore</h1>
        <div>
          <ul>
            {links.map((link) => (
              <li key={link.id}>
                {' '}
                <NavLink
                  to={link.path}
                >
                  {link.text}
                </NavLink>
              </li>
            ))}
          </ul>
        </div>
      </nav>
    );
  }
}
export default Header;
