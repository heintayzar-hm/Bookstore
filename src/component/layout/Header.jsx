import React from 'react';
import { NavLink } from 'react-router-dom';
import { BsFillPersonFill } from 'react-icons/bs';

// eslint-disable-next-line react/prefer-stateless-function
class Header extends React.Component {
  constructor(props) {
    super();
    this.props = props;
  }

  render() {
    const links = [
      {
        id: 1,
        path: '/books',
        text: 'Books',
      },
      {
        id: 2,
        path: '/categories',
        text: 'Categories',
      },
    ];
    return (
      <nav className="border-b-2 border-solid border-[#e8e8e8] items-center justify-center font-important py-[24px] px-[1%] md:px-[4%] grid grid-cols-10 text-center">
        <h1 className="font-bold col-span-4 text-lg sm:text-3xl text-blue">
          <NavLink
            to={links[0].path}
            data-testid="link-item"
          >
            Bookstore CMS
          </NavLink>
        </h1>
        <div className="col-span-5">
          <ul className="grid grid-cols-2 text-sm md:text-base h-full text-center items-center">
            {links.map((link) => (
              <li key={link.id} className="text-brown dark:text-white dark:opacity-100 opacity-50 tracking-widest">
                {' '}
                <NavLink
                  to={link.path}
                  data-testid="link-item"
                  style={({ isActive }) => ({ opacity: isActive ? 1 : 0.5 })}
                >
                  {link.text}
                </NavLink>
              </li>
            ))}
          </ul>
        </div>
        <div className="w-full">
          <BsFillPersonFill className="text-blue hidden xs:block w-8 h-8 p-1 border-b-slate-200 border-2 rounded-full" />

        </div>
      </nav>
    );
  }
}
export default Header;
