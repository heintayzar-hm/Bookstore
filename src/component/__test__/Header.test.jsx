import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { HashRouter } from 'react-router-dom';
import Header from '../Header';

describe('Header', () => {
  test('renders the initial state correctly', () => {
    // Arrange

    // Act
    const { container } = render(
      <HashRouter><Header /></HashRouter>
      ,
    );
    // Assert
    expect(container).toMatchSnapshot();
  });

  test('renders the initial state correctly', () => {
    // Arrange

    // Act
    const { getAllByTestId } = render(
      <HashRouter><Header /></HashRouter>
      ,
    );
    // Assert
    expect(getAllByTestId('link-item')).toHaveLength(2);
  });
});
