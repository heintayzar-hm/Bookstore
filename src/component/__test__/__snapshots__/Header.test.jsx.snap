// Vitest Snapshot v1

exports[`Header > renders the initial state correctly 1`] = `
<div>
  <nav>
    <h1>
      BookStore
    </h1>
    <div>
      <ul>
        <li>
           
          <a
            class=""
            data-testid="link-item"
            href="#/books"
          >
            Books
          </a>
        </li>
        <li>
           
          <a
            class=""
            data-testid="link-item"
            href="#/categories"
          >
            Categories
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
